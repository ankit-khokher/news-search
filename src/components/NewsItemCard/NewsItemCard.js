import "./NewsItemCard.css";

const NewsItemCard = (props) => {
  return <div className="News-article"></div>;
};

export default NewsItemCard;
