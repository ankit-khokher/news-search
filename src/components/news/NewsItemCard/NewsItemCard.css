.News-article {
  margin-top: 20px;
}

.item {
  padding: 1em;
  list-style-type: none;
  border-radius: 6px;
  margin-bottom: 1em;
  background: #fff;
  box-shadow: 0px 2px 6px 2px #0002;
}

.item .title {
  font-size: 100%;
}

.item .content {
  margin: 1em 0;
}

.item .author {
  font-size: 80%;
  font-weight: bold;
  color: #555;
}
