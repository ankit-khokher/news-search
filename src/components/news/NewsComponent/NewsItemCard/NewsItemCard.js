import "./NewsItemCard.css";

const NewsItemCard = (props) => {
  return (
    <div className="News-article">
      {props.articles.map((article, index) => (
        <div className="item" key={index}>
          <h2 className="title">{article.title}</h2>
          <label className="author"> {article.author}</label>
          <p className="content">{article.content.substring(0, 150)}</p>
          <a href={article.url}>{article.url}</a>
        </div>
      ))}
    </div>
  );
};

export default NewsItemCard;
