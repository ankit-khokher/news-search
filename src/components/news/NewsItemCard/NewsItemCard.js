import "./NewsItemCard.css";

const NewsItemCard = (props) => {
  return (
    <div className="News-article">
      {props.articles.map((article, index) => (
        <div className="item" key={index}>
          <h2 className="title">{article.title}</h2>
          <p className="content">{article.content.substring(0, 150)}</p>
          <label className="author"> {article.author}</label>
        </div>
      ))}
    </div>
  );
};

export default NewsItemCard;
