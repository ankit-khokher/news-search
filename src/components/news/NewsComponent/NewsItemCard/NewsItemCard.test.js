import { shallow } from "enzyme";
import NewsItemCard from "./NewsItemCard";

describe("NewsItemCard", () => {
  const props = [
    {
      title: "Showcases Todayâ€™s South Korean Fashion",
      author: "Hahna Yoon",
      content:
        "Ms. Choi credited a Korean television series called Reply, about a group of friends in the 80s and 90s. The three-part drama, one of the most successful shows from 2012 to 2016",
      url:
        "https://www.nytimes.com/2021/04/24/style/minari-fashion-vintage.html",
    },
  ];

  it("should render correctly", () => {
    const wrapper = shallow(<NewsItemCard articles={props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
