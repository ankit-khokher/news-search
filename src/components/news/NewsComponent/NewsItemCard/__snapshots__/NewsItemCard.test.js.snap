// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItemCard should render correctly 1`] = `ShallowWrapper {}`;
